ci: update dependencies in build.env

Madhu-1 opened the pull request:
This commits updates below items.

* Golang version to v1.18.4
* Golangci-lint to v1.47.3
* Helm version to v3.9.2
* Rook version to v1.9.8

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Note:- we have golang 1.19 released yesterday, sticking to the last stable release, for now, If required, we can update to v1.19

Madhu-1 committed:
ci: update dependencies in build.env

This commits updates below items

* Golang version to v1.18.5
* Golangci-lint to v1.47.3
* Helm version to v3.9.2
* Rook version to v1.9.8

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

pkalever commented on build.env:
```diff
@@ -16,14 +16,14 @@ BASE_IMAGE=quay.io/ceph/ceph:v17
 
 # standard Golang options
-GOLANG_VERSION=1.17.10
+GOLANG_VERSION=1.18.5
```
Go 1.19, is a major release. Generally, Major releases are considered stable because they undergo complete testing.
Maybe we should use it?

Madhu-1 replied:
we stick to N-1 release (at least for now) if we all are agreed, we can move to 1.19

cc @ceph/ceph-csi-maintainers

Madhu-1 committed:
ci: disable nosnakecase static check

All the CSI spec imports are having snakecase
we cannot fix it, instead of disabling all imports
skipping the nosnakecase static check.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

Madhu-1 committed:
cephfs: fix staticcheck comment

getting is unused for linter "staticcheck"
(nolintlint) error message due to wrong
comment format. this the format now with
`//directive // comment`

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

nixpanic commented:
#3286 gets merged today, so tomorrow the new container-images will be in the CI mirror. No need to retest/rebase before that.

Madhu-1 commented:
@mergifyio rebase

Madhu-1 commented:
/rest ci/centos/mini-e2e-helm/k8s-1.22

Madhu-1 commented:
/retest ci/centos/mini-e2e-helm/k8s-1.24

Madhu-1 commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

Madhu-1 commented:
@mergifyio rebase

Madhu-1 commented:
@mergifyio rebase
